DEVELOPER_GUIDE.md

```diff
@@ -102,18 +102,19 @@ cd jni
 sed -i -e 's/\/usr\/local\/opt\/libomp\//\/opt\/homebrew\/opt\/llvm\//g' cmake/init-faiss.cmake
 sed -i -e 's/__aarch64__/__undefine_aarch64__/g' external/faiss/faiss/utils/distances_simd.cpp
 sed -i -e 's/pragma message WARN/pragma message /g' external/nmslib/similarity_search/src/distcomp_scalar.cc
-// Change -mcpu value to use chip version according to your M series, for example, -mcpu=apple-m1
+// Change -mcpu value to use chip version according to your M series, for example, -mcpu=apple-m1. You can see the supported mcpu values via gcc --print-supported-cpus
 sed -i -e 's/-march=native/-mcpu=apple-m1/g' external/nmslib/similarity_search/CMakeLists.txt
 sed -i -e 's/-mcpu=apple-a14/-mcpu=apple-m1/g' external/nmslib/python_bindings/setup.py
+sed -i '' '/set(CMAKE_CXX_STANDARD_REQUIRED True)/a\'$'\n''set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xclang -fopenmp -L/opt/homebrew/opt/libomp/lib -I/opt/homebrew/opt/libomp/include -lomp -arch arm64 -fexceptions")' CMakeLists.txt
 
 // Install llvm
 brew install llvm
 echo 'export PATH="/opt/homebrew/opt/llvm/bin:$PATH"' >> ~/.zshrc
 source ~/.zshrc
 
 // Set compiler path for CMAKE
-export CC=/opt/homebrew/opt/llvm/bin/clang
-export CXX=/opt/homebrew/opt/llvm/bin/clang++
```

Comment:
This works for me. Can we keep both? you can add a note saying change to 
```
in case of error: [Add the error] change to
export CC=gcc
export CXX=g++
```

Reply:
Sure and let me ask Andrew what his error was -- I think they were different.

Reply:
Pasting entire error trace here so that people can find it in search. Both of us had linking errors with clang: 
```
<after running make>
77%] Building CXX object CMakeFiles/opensearchknn_faiss.dir/src/faiss_wrapper.cpp.o
[ 79%] Building CXX object CMakeFiles/opensearchknn_faiss.dir/src/faiss_util.cpp.o
<FAILS HERE> [ 79%] Linking CXX shared library release/libopensearchknn_faiss.jnilib
```
Here's the trace: 
```
Homebrew clang version 18.1.7
Target: arm64-apple-darwin23.5.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
 "/usr/bin/ld" -demangle -lto_library /opt/homebrew/Cellar/llvm/18.1.7/lib/libLTO.dylib -dynamic -dylib -arch arm64 -dylib_install_name @rpath/libopensearchknn_faiss.jnilib -platform_version macos 14.0.0 14.4 -syslibroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -mllvm -enable-linkonceodr-outlining -o release/libopensearchknn_faiss.jnilib -headerpad_max_install_names CMakeFiles/opensearchknn_faiss.dir/src/org_opensearch_knn_jni_FaissService.cpp.o CMakeFiles/opensearchknn_faiss.dir/src/faiss_wrapper.cpp.o CMakeFiles/opensearchknn_faiss.dir/src/faiss_util.cpp.o -rpath /Users/finnrobl/Code/k-NN/jni/release external/faiss/faiss/libfaiss.a release/libopensearchknn_util.jnilib -framework Accelerate -lm -ldl /opt/homebrew/opt/llvm/lib/libomp.dylib -lc++ -lSystem /opt/homebrew/Cellar/llvm/18.1.7/lib/clang/18/lib/darwin/libclang_rt.osx.a
Undefined symbols for architecture arm64:
  "std::exception_ptr::__from_native_exception_pointer(void*)", referenced from:
      std::__1::promise<bool>::~promise() in libfaiss.a[77](WorkerThread.cpp.o)
  "___cxa_init_primary_exception", referenced from:
      std::__1::promise<bool>::~promise() in libfaiss.a[77](WorkerThread.cpp.o)
ld: symbol(s) not found for architecture arm64
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [release/libopensearchknn_faiss.jnilib] Error 1
make[1]: *** [CMakeFiles/opensearchknn_faiss.dir/all] Error 2
make: *** [all] Error 2
```